pkg/bundle/service.go

```diff
@@ -46,11 +47,26 @@ func (s BundleConfigurationService) Configure(ctx context.Context, cluster *clus
 	logger.Info("configuring observability-bundle")
 
 	bundleConfiguration := bundleConfiguration{
-		Apps: map[string]app{
-			"prometheusAgent": {
-				Enabled: s.config.IsMonitored(cluster),
-			},
-		},
+		Apps: map[string]app{},
+	}
+
+	switch s.config.MonitoringAgent {
+	case common.MonitoringAgentPrometheus:
+		bundleConfiguration.Apps[common.MonitoringPrometheusAgentAppName] = app{
+			Enabled: s.config.IsMonitored(cluster),
```

Comment:
I think this Can bé simplified without the Switch with: s.config.IsMonitored(cluster) && common.MonitoringAgentPrometheus == s.config.MonitoringAgent. this would definitely reduce duplication

Reply:
If you want to keep the Switch, maybe move the bundleConfiguration.Apps outside thé Switch?

Reply:
You could use a function liké enableApp or disabledApp instead?

Reply:
This implementation make it clear what is happening each case. I don't see how I can move bundleConfiguration outside the switch. Also I would rather not add a function which would obfuscate the behavior here.
If you feel like making a suggestion do it via github suggestion because I am having a hard time to grasp what you have in mind.

Reply:
Well, i'll try 😅

Reply:
So, i'm fine with thé Switch per se but I think this validation should be in the main go file and not here as the operator should panic if the logging agent does not work. This would definitely simplify the code here

Reply:
I'm not sure using a correctly names function obfuscates the code though. The original reason why I brought this up is that the logging operator will bé merged in the observability operator and this is also the place where we would enabled/disabled logging agents on the future but we can do without thé function today

Reply:
Having a validation in main.go would be nice and allow to fail early, but here we would still need to handle all possible different values for the logging agent.

Merging the observability and logging operator would require major refactoring so I would rather leave this for later.

Reply:
Well if you add validation in main.go then you can change thé metricAgent type to an enum type with only 2 allowed values, alloy and prometheusAgent so you don't need a Switch but an if else right?

Reply:
I added the check to ensure observability bundle support alloy as monitoring agent.

I rather keep the switch as is for now and not add enum, it's safer with enum true, but still we have a case where multiple values are possible and I prefer to handle all cases including when the value is not what's expected.

Reply:
Let's go with that for now then :)